test/unit/auth_sources/auth_source_ldap_test.rb

```diff
@@ -167,7 +167,16 @@ def setup
     end
 
     test 'update_usergroups calls refresh_ldap if entry belongs to some group' do
-      ExternalUsergroup.expects(:find_by_name).with('ipausers').returns(ExternalUsergroup.new)
+      @auth_source_ldap.expects(:external_usergroups).returns(ExternalUsergroup)
+      ExternalUsergroup.expects(:where).with('lower(name) = ?', 'ipausers').returns([ExternalUsergroup.new])
```

Comment:
Since we're using a little bit of SQL here, can you avoid stubbing it and instead use the real DB?  This'll mean we get testing on all three supported DBs via unit tests.